src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -277,6 +342,60 @@ public void sent(long size) {
         }
 
         @Override
+        public void onRequestSent(DiscoveryNode node, long requestId, String action, TransportRequest request, TransportRequestOptions options) {
```

Comment:
I think we should have a `protected void trace(String, Object...)` method here that we can override and then simply test with an SingleNodeTest. I really don't like the fact that we are capturing log lines in order to test this feature